### PR TITLE
Added functionality to save credentials for private repos.

### PR DESCRIPTION
**Pull request by satishc1**
Fix for https://github.com/capitalone/Hygieia/issues/654

picture1

### CODE REVIEW

**Comment by MarkRx** [comment]
Would it be possible to add a warning to the UI & API layers if TLS/SSL isn't set up?

**Comment by satishc1** [comment]
Yes.. this is being planned. Added https://github.com/capitalone/Hygieia/issues/836 to reference this.

**Comment by dinukool** [comment]
will this work with Stash private repo -- Bitbucket collector?

**Comment by satishc1** [comment]
@dinukool I've tested it with Github. But it should work with Bitbucket also.

File: UI/src/components/widgets/repo/config.html [hunk 1]
```diff
@@ -48,6 +48,29 @@
...
+		<input type="text"
+			   name="repouser"
+			   class="form-control text-center"
+			   ng-model="repoConfig.repouser"
+			   maxlength="25"
+			   autocomplete="off"
+			   placeholder="Enter your repo username"/>
+	</form-group>
+	<form-group input="repopass"
+				errors="{errorEncryptingPassword: 'Encryption error. Verify key in api.properties and try again.'}">
```
Comment (tabladrum) [inline UI/src/components/widgets/repo/config.html]: Lets keep the error message simple. Not talk about api.properties etc.

File: UI/src/components/widgets/repo/config.js [hunk 2]
```diff
@@ -68,25 +66,44 @@
...
-							processCollectorItemResponse);
-				} else if (repoType == 'GitHub (private)') {
-					var httpReplace = ("http://").concat(ctrl.username).concat(
-							":").concat(ctrl.password).concat("@");
-					var httpsReplace = ("https://").concat(ctrl.username)
-							.concat(":").concat(ctrl.password).concat("@");
-					var url2 = url.replace("http://", httpReplace);
-					var url3 = url2.replace("https://", httpsReplace);
-					createCollectorItem(url3, repoType, ctrl.gitBranch).then(
-							processCollectorItemResponse);
```
Comment (tabladrum) [inline UI/src/components/widgets/repo/config.js]: Thanks for cleaning this up.

**Comment by tabladrum** [comment]
LGTM

**Comment by amitmawkin** [comment]
LGTM

**Comment by tabladrum** [comment]
LGTM

**Comment by amitmawkin** [comment]
LGTM

**Comment by oshingc** [comment]
Authentication is not working by the UI for Stash (Bitbucket Collector)?

**Comment by oshingc** [comment]
:c

**Comment by anuragmishracrest** [comment]
@satishc1 How to generate key for private repo.

**Comment by niharamin** [comment]
@satishc1  This functionalist doesn't work for bitbucket.  I keep getting unauthorized error.  i changed my password to something very simple and it still gives me unauthorized error. 

ERROR c.c.d.c.DefaultBitbucketCloudClient - 404 Not Found:https://bitbucket.org/api/2.0/repositories

**Comment by satishc1** [comment]
@niharamin Sure it does. The error log indicates you have not added the key to your bitbucket properties file.

image

**Comment by niharamin** [comment]
i confirm it working. Thank you

INFO  c.c.d.collector.CollectorTask - Running Collector: Bitbucket
INFO  c.c.d.collector.CollectorTask - -----------------------------------
INFO  c.c.d.collector.CollectorTask - Starting...
INFO  c.c.d.collector.CollectorTask - -----------------------------------
INFO  c.c.d.collector.CollectorTask - Repo Count              2        0s
INFO  c.c.d.collector.CollectorTask - New Commits             0        0s
INFO  c.c.d.collector.CollectorTask - Finished                         0s

**Comment by sri1256** [comment]
I am successfully running github .but values are not coming on dashboard.
github-collector: application.properties file
dbname=dashboard
dbusername=dashboarduser
dbpassword=dbpassword
dbhost=localhost
dbport=27017
dbreplicaset=false
dbhostport=localhost:27017,127.0.0.1:27017
server.port=8082
github.cron=0 0/5 * * * *

github.host=https://github3.cisco.com
github.UserId=<githubusername>
github.key=UQt6YUxFQ5f7wRnaEK2MQAdzH3UvdqTv(my generated secret key)
github.personalAccessToken=<github-personal_access-key>
#Maximum number of days to go back in time when fetching commits
github.commitThresholdDays=15

api dashboard.properties
dbname=dashboard
dbusername=dashboarduser
dbpassword=dbpassword
dbhost=localhost
dbport=27017
dbreplicaset=false
dbhostport=localhost:27017,127.0.0.1:27017
server.contextPath=/api
server.port=8080
corsEnabled=false
#corsWhitelist=http://domain1.com:port,http://domain2.com:port
#version.number=@application.version.number@
key=UQt6YUxFQ5f7wRnaEK2MQAdzH3UvdqTv
auth.expirationTime=7200000
#auth.secret=
auth.secret=hygieiasecret
auth.authenticationProviders=STANDARD


screenshot-2017-12-15 f9faf33f-08d7-4d4f-b020-f48d05b2c7bc png png image 2868 x 1324 pixels - scaled 58

Without creditionals
screenshot-2017-12-15 hygieia dashboard
 

Anyone give me reply soon

**Pull request merged**